### ISSUE TITLE
Random Steering Loadbalancer Weights gets set to default when updating

### ISSUE THREAD

**Issue by jeff-onepeloton** [cloudflare/terraform-provider-cloudflare#2641]
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

4.18

### Affected resource(s)

cloudflare_load_balancer

### Terraform configuration files

```hcl
resource "cloudflare_load_balancer" qa1_cms_stacks {
  default_pool_ids = [data.cloudflare_load_balancer_pools.qa1_cms_stack_a.pools[0].id, data.cloudflare_load_balancer_pools.qa1_cms_stack_b.pools[0].id]
  enabled          = true
  fallback_pool_id = data.cloudflare_load_balancer_pools.qa1_cms_stack_a.pools[0].id
  name             = "qa1-cms-stacks.onepeloton.com"
  proxied          = true
  session_affinity = "none"
  steering_policy  = "random"
  zone_id          = data.cloudflare_zone.zone.id
  adaptive_routing {
    failover_across_pools = false
  }
  location_strategy {
    mode       = "pop"
    prefer_ecs = "proximity"
  }
  random_steering {
    default_weight = 1
    pool_weights = {
      (data.cloudflare_load_balancer_pools.qa1_cms_stack_a.pools[0].id) = 0
      (data.cloudflare_load_balancer_pools.qa1_cms_stack_b.pools[0].id) = 1
    }
  }
}
```

### Link to debug output

Its in the slack we share (onepeloton)

### Expected output

I expected the weights to shift from 0/1 to 1/0 from one pool to the other.

### Actual output

The weights became 0.5 / 0.5 between the two. I think what it did is set both of them to be the default weight but I'm not sure.

### Steps to reproduce

Create a loadbalancer with random weight steering between two pools. Then try to change the weights.

### References

Think it was briefly fixed in provider version 4.5 (and I can confirm it was working):
https://github.com/cloudflare/terraform-provider-cloudflare/pull/2403

But then I think the issue started reappearing again when a newer version got released.

**Comment by jacobbednarz** [comment]
> Its in the slack we share (onepeloton)

please provide the debug output here. I don't have access and cannot triage this for you. 

> Think it was briefly fixed in provider version 4.5 (and I can confirm it was working):
#2403

#2405 didn't fix the issue it set out to. the issue was flakey in nature so it would have appeared to be addressed but not actually solved it. 

have you tried the latest master [dev build](https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/contributing/using-non-released-versions.md)? #2635 fixed the issue with random steering values however, I'm unable to confirm if that is your issue given the lack of debug output here.

**Comment by jacobbednarz** [comment]
closing this off as addressed by #2635. if it's not please provide the debug output in a fresh issue that reproduces on the latest version and we can investigate further.
